app/src/main/java/com/connorlay/knitgrid/presenters/PatternPresenter.java

```diff
@@ -112,6 +113,15 @@ public Long getPatternId() {
         return mPatternId;
     }
 
+    public void setStitchColor(int colorID, int row, int col){
+        if (mStitchGrid == null){
```

Comment:
I think this guard statement should instead be if (mStitchGrid[row][column] == null) return;